### PR TITLE
feat: show alert when rendering a note with unsafe content

### PR DESCRIPTION
**Pull request by johnny243**

### CODE REVIEW

File: src/main.js [hunk 1]
```diff
@@ -30,6 +30,35 @@ document.addEventListener('DOMContentLoaded', function() {
...
+    } else {
+      renderNote = true;
+    }
+
+    /**
+       * If the user decides not to continue rendering the note,
+       * clear the editor and disable it.
+       */
+    if (!renderNote) {
+      window.easymde.value('');
```
Comment (atmoio) [inline src/main.js]: Will this overwrite the note's contents with an empty string and save it? Or something else?

File: src/main.js [hunk 2]
```diff
@@ -171,4 +200,61 @@ document.addEventListener('DOMContentLoaded', function() {
...
+    const notesPreferences = getNotePreferences();
+    return notesPreferences[lastUUID];
+  }
+
+  function setNotePreferences(key, value) {
+    const notesPreferences = getNotePreferences();
+    notesPreferences[lastUUID] = {
+      [key]: value
+    };
+    componentRelay.setComponentDataValueForKey('notes', notesPreferences);
```
Comment (atmoio) [inline src/main.js]: Hmm, why are we not setting this preference on the note item itself via `clientData`?

Comment (johnny243) [reply src/main.js]: The only downside I see of using the `clientData` is that if the note is duplicated, the alert will not be shown again (correct me if `clientData` is also kept when duplicating a note). Do other domains exists in `clientData` (aside from the default one)?

Comment (atmoio) [reply src/main.js]: I think it's on the contrary: if a note is duplicated, its clientData is duplicated. However if you save the preference on the component itself (as is being done now), and the note is duplicated, then this data won't carry over.

Comment (johnny243) [reply src/main.js]: Ah, got it. Will use `clientData` then 👍🏽

File: src/main.js (hunk 1 above)
Comment (johnny243) [reply src/main.js]: It will set the editor's text to a blank string, but rendering and saving will be disabled.

**Review by atmoio — APPROVED** [review]

**Pull request merged**